runtime/lua/vim/lsp/util.lua

```diff
@@ -373,9 +373,12 @@ function M.compute_diff(old_lines, new_lines, start_line_idx, end_line_idx)
     adj_end_char = #old_lines[#old_lines + end_line + 1] + end_char + 1
   end
 
+  start_char = vim.str_utfindex(old_lines[start_line], start_char - 1)
```

Comment:
It should be the second return value for `vim.str_utfindex` for UTF-16 (the difference will only be noticable with non-BMP chars)

Reply:
https://github.com/mjlbach/neovim/pull/new/fix_utf16_ranges_again